kythe/go/extractors/config/smoke/smoke.go

```diff
@@ -103,15 +127,31 @@ func (g harness) TestRepo(ctx context.Context, repo string) (Result, error) {
 	fromRepo, err := g.filenamesFromRepo(ctx, repo)
 	if err != nil {
 		log.Printf("Failed to read repo from remote: %v", err)
-		return Result{false, false, len(fromRepo), 0, 0.0}, nil
+		return Result{
+			Downloaded:    false,
+			Extracted:     false,
+			Indexed:       false,
+			DownloadCount: len(fromRepo),
+			ExtractCount:  0,
+			IndexCount:    0,
+			FileCoverage:  0.0,
+		}, nil
 	}
 
 	fromExtraction, err := g.filenamesFromExtraction(ctx, repo)
 	if err != nil {
 		log.Printf("Failed to extract repo: %v", err)
 		// TODO(danielmoy): consider handling errors independently and
 		// returning separate false results if either err != nil.
-		return Result{true, false, len(fromRepo), len(fromExtraction), 0.0}, nil
+		return Result{
+			Downloaded:    true,
+			Extracted:     false,
+			Indexed:       false,
+			DownloadCount: len(fromRepo),
+			ExtractCount:  len(fromExtraction),
+			IndexCount:    0,
+			FileCoverage:  0.0,
```

Comment:
`FileCoverage: 0`

Untyped literals get concrete type from their use, it's not necessary to decorate them lexically (in most cases).

https://blog.golang.org/constants, if you're interested in more.

Reply:
Ah right, forgot about that.

Reply:
It depends. It's generally idiomatic to rely on default zero values for things—and in fact it's a good idea to choose field names so that the zero _does_ make a sensible default, where practical. But sometimes it can be worthwhile for documentation. Here, I probably wouldn't bother except maybe to document that the default coverage allows anything to be "ok".